### DISCUSSION TITLE
Padding around <Link> not behaving as wanted

### DISCUSSION THREAD

**Discussion by skaidigitalnorge** [vercel/next.js#32162] [Help]
Hello! I've had an issue with styling the <Link> tag for a while, and thought I'd reach out for help.

Whenever I use the Link-tag as a button and give the inner <a>-tag padding, it doesn't seem like the rest of the UI "understands" that it has padding. In the image below, can image is set to follow the right of the text-div on the right. The padding doesn't expand the image height, and the text above is not pushed further away either

![2021 12 06](https://user-images.githubusercontent.com/92500207/144815934-f4001b24-f409-45dd-8349-c29fdf38c795.png)
<Link>-tag with padding in the inner <a>-tag

If I remove the padding, the distance between all the elements stay the same, as you can see in the image below

![2021 12 06](https://user-images.githubusercontent.com/92500207/144815943-a12a87f3-0cc5-4513-8b0d-4d406b708e75.png)
Same content with the padding removed

Here is the code for the button. The same issue occurs even if I remove all the inner spans.

`const PrimaryButtonStyling = `justify-center bg-cta border-2 border-cta text-hvit font-semibold w-0100 md:w-auto bg-primary_700 p- text-white`;
export const PrimaryButtonHref = ({ text, href, iconLeft, iconRight }) => {
  return (
    <Link href={href}>
      <a className={`${PrimaryButtonStyling}`}>
        {iconLeft != null ? <span className="mr-16">{iconLeft}</span> : null}
        {text}
        {iconRight != null ? <span className="ml-16">{iconRight}</span> : null}
      </a>
    </Link>
  );
};`

![2021 12 06](https://user-images.githubusercontent.com/92500207/144816111-90d899fe-71b5-452d-9f6f-c556feab5717.png)

Any help is greatly appreciated!

**Comment by skaidigitalnorge** [comment]
Update. I added flex to the class, and it seems to have resolved the issue!

**Reply by xjanse** [reply]
I had the same issue. Thanks for the solution!

**Reply by hazeled** [reply]
Same issue here, and that fixes it. Why does this work??

**Comment by szkieb** [comment]
![image](https://user-images.githubusercontent.com/102992385/218834400-e6455c69-881c-4ba1-8610-0c1678e0a579.png)

Running next v13.1.6 and encountered the same issue. 
The padding is applied to the Link element, rendered as anchor element. But the element's position remains the same as without padding so that it overlaps and covers the paragraph above. 

Adding flex to Link element's list of classes did solve the problem.
